flang/lib/Lower/OpenMP.cpp

```diff
@@ -75,6 +75,35 @@ genOMP(Fortran::lower::AbstractConverter &absConv,
       standaloneConstruct.u);
 }
 
+static void
+genOMP(Fortran::lower::AbstractConverter &absConv,
+       Fortran::lower::pft::Evaluation &eval,
+       const Fortran::parser::OpenMPBlockConstruct &blockConstruct) {
+  const auto &blockDirective =
+      std::get<Fortran::parser::OmpBeginBlockDirective>(blockConstruct.t);
+  const auto &parallelDirective =
+      std::get<Fortran::parser::OmpBlockDirective>(blockDirective.t);
+  if (parallelDirective.v == llvm::omp::OMPD_parallel) {
+    auto &firOpBuilder = absConv.getFirOpBuilder();
+    auto currentLocation = absConv.getCurrentLocation();
+    auto insertPt = firOpBuilder.saveInsertionPoint();
+    llvm::ArrayRef<mlir::Type> argTy;
+    mlir::ValueRange range;
+    std::vector<int32_t> operandSegmentSizes(6);
```

Comment:
1) Can the size of the vector be found from the parallel operation?
2) Can llvm small vector be used? What is typically used in mlir std::vector or llvm vector?

Reply:
llvm::SmallVector is a good suggestion, since we know the exact size at compile time.

Reply:
Used SmallVector, as it's well in present situation. mlir coding convention is flexible for both SmallVector and std::vector on need basis.
Thanks!